### ISSUE TITLE
inline fun <reified T : Plugin<*>> PluginAware.apply(to: Any) is error prone

### ISSUE THREAD

**Issue by mkobit** [gradle/kotlin-dsl#1161]
### Expected Behavior

Alignment of plugin application so there are aren't so many ways to do it.

The signature looks to be:

```kotlin
inline fun <reified T : Plugin<*>> PluginAware.apply(to: Any) {
    apply {
        it.plugin(T::class.java)
        it.to(to)
    }
}
```

Should be one of:

* a compile time failure by expecting a `PluginAware` parameter
* a compile time failure by not specifying the generic `Plugin` type (don't know why it doesn't fail now
* removed from API

### Current Behavior

Fails at runtime with something like

```
Cannot apply plugin of class 'org.gradle.api.Plugin' to 'base' (class: java.lang.String) as it does not implement PluginAware
```

### Context

* Coworkers unfamiliar with `kotlin-dsl` ran into this
* I ran into this unintentionally when preparing a presentation
* https://stackoverflow.com/questions/52563215/gradle-settings-groovy-file-to-kts-file

### Steps to Reproduce (for bugs)

```kotlin
apply("base")
```

### Your Environment

Gradle 4.10.2

**Comment by eskatos** [comment]
Thanks for the report @mkobit, agreed it is confusing.
What about renaming it to `applyTo<T>(target)` to remove the ambiguity, wdyt @bamboo?

**Comment by eskatos** [comment]
Closing as fixed by #1184
